doc/man1/openssl-verification-options.pod

```diff
@@ -151,20 +173,21 @@ The first step is to check that each certificate is well-formed.
 Part of these checks are enabled only if the B<-x509_strict> option is given.
 
 The second step is to check the extensions of every untrusted certificate
-for consistency with the supplied purpose.
-If the B<-purpose> option is not given then no such checks are done
-except for SSL/TLS connection setup,
-where by default C<sslserver> or C<sslclient>, are checked.
-The target or "leaf" certificate, as well as any other untrusted certificates,
-must have extensions compatible with the specified purpose.
+for consistency with any given purpose.
+If the B<-purpose> option is not given
+then no such checks are done except for SSL/TLS connection setup,
+where by default C<sslserver> or C<sslclient> is checked.
+Each certificate must either not have an EKU extension,
+or the EKU extension must include B<anyExtendedKeyUsage>
+or the given required purpose.
 All certificates except the target or "leaf" must also be valid CA certificates.
 The precise extensions required are described in more detail in
 L<openssl-x509(1)/CERTIFICATE EXTENSIONS>.
 
 The third step is to check the trust settings on the last certificate
 (which typically is a self-signed root CA certificate).
-It must be trusted for the given use.
-For compatibility with previous versions of OpenSSL, a self-signed certificate
+It must be trusted for the given use (if specified).
```

Comment:
I'd write this instead as:

_If specified, it must be trusted for the given use._